org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/builderState/AbstractBuilderState.java

```diff
@@ -68,17 +70,26 @@
 	private WorkspaceLockAccess workspaceLockAccess;
 
 	private volatile boolean isLoaded = false;
+	
+	final ILock lock = Job.getJobManager().newLock();
 
 	public void load() {
 		if (!isLoaded) {
 			IWorkspaceRunnable runnable = new IWorkspaceRunnable() {
 				@Override
 				public void run(IProgressMonitor monitor) throws CoreException {
 					if (!isLoaded) {
-						resourceDescriptionData = new ResourceDescriptionsData(persister.load());
-						if(storage2UriMapper instanceof IStorage2UriMapperExtension)
-							((IStorage2UriMapperExtension) storage2UriMapper).initializeCache();
-						isLoaded = true;
+						try {
+							lock.acquire();
```

Comment:
If I'm not mistake, the lock should be aquired before the try block. Otherwise we'll try to release something in the finally block, if the acquisition failed from some reason.

Reply:
Hmm weird, the documentation for ILock has an example that states otherwise...